practice-one/src/pages/Analytics/Analytics.jsx

```diff
@@ -1,7 +1,68 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import "./Analytics.css";
+import { Button, Expand, CustomerItem } from "@components";
+import { BUTTON_VARIANTS } from "@constants/buttons";
+import { plusIcon } from "@assets/images";
+import { EXPAND_TITLES } from "@data";
+import { getAllCustomerService } from "../../services/customerService";
+import { v4 as uuidv4 } from "uuid";
 
 const Analytics = () => {
-  return <h1>Analytic</h1>;
+  const [customers, setCustomers] = useState([]);
+
+  useEffect(() => {
+    const getAllCustomers = async () => {
+      // Call getAllCustomerService function to get all data
+      const data = await getAllCustomerService();
+      setCustomers(data);
+    };
+    getAllCustomers();
+  }, []);
+
+  const renderList = () => {
```

Comment:
what is list will render?

Reply:
It will return a array having data of all customers, then I added it to customers array.
After that, I call customer array to show each customer's info on page.

Reply:
so renderList is it doesn't make sense

Reply:
No, this function helps create a dynamic interface list based on available customer data in the customers array, helping to display each customer's information in the website interface.

Reply:
please detail the list will render, renderList is the bad name for function

Reply:
I renamed this function
![image](https://github.com/datdragon777/React-Training/assets/74806826/74cc46a6-059e-4c7e-8604-9364bd8aa090)